gufe/components/proteincomponent.py

```diff
@@ -51,23 +51,78 @@
 _CHIRALITY_STR_TO_RDKIT = {v: k for k, v in _CHIRALITY_RDKIT_TO_STR.items()}
 
 
-negative_ions = ["F", "CL", "BR", "I"]
-positive_ions = [
-    # +1
-    "LI",
-    "NA",
-    "K",
-    "RB",
-    "CS",
-    # +2
-    "BE",
-    "MG",
-    "CA",
-    "SR",
-    "BA",
-    "RA",
-    "ZN",
-]
+ions_dict = {
+    # Alkali metals
+    "LI": 1,
+    "NA": 1,
+    "K": 1,
+    "RB": 1,
+    "CS": 1,
+    "K+": 1,
+    "Na+": 1,
```

Comment:
I don't think names which are a combination of upper and lower characters can be matched as we are getting the ion name as upper characters only. So this list needs to all be in upper case.